test/python/circuit/library/test_blueprintcircuit.py

```diff
@@ -209,6 +209,34 @@ def _build(self):
             math.pi / 2,
         )
 
+    def test_copy_empty_like(self):
+        """Test copying empty-like.
+
+        Regression test of #13535.
+        """
+        circuit = EfficientSU2(2)
+        circuit.global_phase = -2
+        circuit.metadata = {"my_legacy": "i was a blueprintcircuit once"}
+
+        cpy = circuit.copy_empty_like()
+        expected = QuantumCircuit(
+            circuit.num_qubits, global_phase=circuit.global_phase, metadata=circuit.metadata
+        )
+        self.assertEqual(cpy, expected)
+        self.assertNotIsInstance(cpy, BlueprintCircuit)
```

Comment:
I don't think the metadata is included in the equality check, so I don't think we'll actually be testing that.

I think we need to trigger the build of `circuit` and do something that would trigger the build of `cpy` (if it were a `BlueprintCircuit`) as well like adding a single gate to it - the `not isinstance` check is ok, but it's a round-about way of testing the behaviour we actually want, which is "the circuit should actually be empty".

Reply:
The `assertEqual` check should trigger the build -- with the old version of the `copy_empty_like` that test failed 🙂 The final instance check was just to be very clear it is no longer a blueprint circuit 😛 I can add a method call that would trigger the build explicitly, if it were a blueprint circuit

Reply:
(I answered the other chain before I read this one.)

Ok, if `assertEqual` did actually trigger the build, then I guess it is testing it.  I guess my now-implied point is that it wasn't obvious to me that `assertEqual` would necessarily have triggered the build, so it wasn't 100% clear to me that the test was correct, and having that certainty is a nice property of tests.

Reply:
I'll add a bit to ensure it would trigger the building and add a comment 🙂